templates/google-login-button.php

```diff
@@ -6,18 +6,33 @@
  * @since 1.0.0
  */
 
-$button_text = ( ! empty( $button_text ) ) ? $button_text : __( 'Log in with Google', 'login-with-google' );
+if ( isset( $custom_btn_text ) && $custom_btn_text ) {
+	$button_text = esc_html( $custom_btn_text );
+} else {
+	$button_text = ( ! empty( $button_text ) ) ? $button_text : __( 'Log in with Google', 'login-with-google' );
+}
+
+if ( is_user_logged_in() ) {
+	$button_text = __( 'Log out', 'login-with-google' );
+}
 
 if ( empty( $login_url ) ) {
 	return;
 }
-
 ?>
 <div class="wp_google_login">
-	<div class="wp_google_login__button-container">
-		<a class="wp_google_login__button" href="<?php echo esc_url( $login_url ); ?>">
-			<span class="wp_google_login__google-icon"></span>
-			<?php echo esc_html( $button_text ); ?>
-		</a>
-	</div>
+    <div class="wp_google_login__button-container">
+        <a class="wp_google_login__button"
+            <?php
+            if ( ! is_user_logged_in() ) {
+                printf( ' href="%s"', esc_url( $login_url ) );
+            } else {
+                printf( ' href="%s"', esc_url( wp_logout_url( get_permalink() ) ) );
+            }
```

Comment:
We are already checking if the user is logged in, on line 15, can't we form the URL there, and use just one printf statement?